Event Page Progress 🏔

EthanThatOneKid opened the pull request:
We created the layout for the `/events` page and rendered the ICAL data from the Google Calendar.

We also fixed some styles that conflicted with the event card that originated in the `call-to-action.svelte` component, so we added those fixes where needed.

- [x] We need a button on either side to increment the scroll-offset slightly in its respective direction.
- [x] We need to make it more mobile-friendly, so let's add some media queries.
- [x] Create a drag gesture/interaction to increment the scroll-offset smoothly. (<https://github.com/EthanThatOneKid/acmcsuf.com/pull/91/commits/1bf4961b34369e20e8c27f782bf027a283bcb807>)
- [x] Automatically scroll to event with the URL's _fragment_.
- [x] Hide buttons for mobile dimensions.
- [x] Hide the scrollbar.

EthanThatOneKid committed:
Here is a rough draft for adding the iCal data to the `/event` page.

I added a new helper function to parse the text data into a list of formatted event interfaces to be read by the Svelte component. This is the beginning of tackling issue #75.

EthanThatOneKid committed:
Update parse-ical-data.ts

Fixed indentation typo.

EthanThatOneKid committed:
Refactored the parsing algorithm

Sorted, formatted, and filtered the AcmEvent objects.

EthanThatOneKid committed:
Initial work on the `/events` page 😎

We created the layout for the `/events` page and rendered the ICAL data from the Google Calendar.

We also fixed some styles that conflicted with the event card that originated in the `call-to-action.svelte` component, so we added those fixes where needed.

EthanThatOneKid committed:
Merge branch 'main' into feature/event-page-preload

EthanThatOneKid committed:
Update event-item.svelte

EthanThatOneKid committed:
Implemented scrolling the carousel via grabbing ✊

When the user clicks and moves their cursor along the event carousel component, the scroll offset will automatically increment. Below is an emoji diagram of the expected gesture.

```
🖐✊➡🖐
```

diamondburned commented on src/components/events/event-carousel.svelte:
```diff
@@ -1,13 +1,52 @@
   import EventItem from "./event-item.svelte";
   import type { AcmEvent } from "../../lib/parse-ical-data";
 
+  enum GrabbingState {
```
Do we need this? Why can't we do `grabbing bool`?

EthanThatOneKid replied:
I am using this enum because it is being used as an attribute value. I assume there is a simpler way to do this involving a boolean. In the end, the CSS just needs to target something to be able to give it the appropriate style which could be a class or an attribute.

We could possibly instead use an `isGrabbing` variable and use it to assign a class onto the div using a Svelte class directive like this: `<div class:grabbing={isGrabbing} ... />`.

EthanThatOneKid replied:
Addressed in commit <https://github.com/EthanThatOneKid/acmcsuf.com/pull/91/commits/b2d9ffd220775905eb63197f599820092cbffe6f>.

EthanThatOneKid committed:
Refactored `grabbing` attribute into CSS class.

Instead of reading the DIV's attributes to handle the grabbing state, I moved it into a variable `isGrabbing` to make the code cleaner.

Suggested from: <https://github.com/EthanThatOneKid/acmcsuf.com/pull/91#discussion_r619888248>

EthanThatOneKid committed:
Merge branch 'main' into feature/event-page-preload
